src/v/cluster/feature_backend.cc

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "feature_backend.h"
+
+#include "seastar/core/coroutine.hh"
+
+namespace cluster {
+
+ss::future<std::error_code>
+feature_backend::apply_update(model::record_batch b) {
+    std::variant<feature_update_cmd> cmd = co_await cluster::deserialize(
+      std::move(b), accepted_commands);
+
+    co_await ss::visit(
```

Comment:
nit: why not to use something like `cmd.get<feature_update_cmd>()...`?

Reply:
Yeah, that's neater.  I think when I first wrote this I hadn't decided how many message types I would use.  Updated.